service/telemetry.go

```diff
@@ -123,6 +136,10 @@ func (tel *telemetryInitializer) initOnce(buildInfo component.BuildInfo, logger
 		telAttrs[semconv.AttributeServiceVersion] = buildInfo.Version
 	}
 
+	if tel.registry.IsEnabled(allowTraceContextPropagationFeatureGateID) {
+		otel.SetTextMapPropagator(autoprop.NewTextMapPropagator())
```

Comment:
```suggestion
		if _, exists := os.LookupEnv("OTEL_PROPAGATOR); !exists && cfg.Traces.Propagator != "" {
			// handle the error, though
			_  = os.SetEnv("OTEL_PROPAGATOR", cfg.Traces.Propagator)
		}
		otel.SetTextMapPropagator(autoprop.NewTextMapPropagator())
```

Something like this could potentially be used to enable control of the selector propagators via config if it isn't already set via environment.

Reply:
@Aneurysm9 can we provide a factory method to construct the right propagators from comma separated strings in the autoprop module -(which I think is the right place). Setting environment variables from within the process doesn't seem clean and side effect-y

IMHO env variables is something that operators should drive. I suggest we document OTEL_PROPAGATOR and introduce configuration `cfg.tracer.Propgators` only when auto prop adds the factory method. Besides having the factory method take in the same input as what gets passed to OTEL_PROPAGATOR makes the module more consistent. 

cc: @mx-psi  @jpkrohling

Reply:
Had a chat with @Aneurysm9 - We have decided to leave the code as is with autoprop and will let users configure it via OTEL_PROPAGATORS. Once the id based factory api is available, will intoduce collector telemetry.traces.propagators configuration.